httpcache.go

```diff
@@ -51,7 +55,48 @@ func CachedResponse(c Cache, req *http.Request) (resp *http.Response, err error)
 	}
 
 	b := bytes.NewBuffer(cachedVal)
-	return http.ReadResponse(bufio.NewReader(b), req)
+	returnResponse, err := http.ReadResponse(bufio.NewReader(b), req)
+	if err != nil {
+		return nil, fmt.Errorf("error loading response from cache: %s\n", err.Error())
+	}
+
+	rangeRaw := req.Header.Get("range")
+	if rangeRaw != "" {
+		tmp := strings.Split(rangeRaw, rangeTypeSeparator)
+		// standard format is bytes=START-END
+		rangetype, rangevalue := tmp[0], tmp[1]
+		if rangetype != "bytes" {
+			fmt.Printf("range type %s not supported\n", rangetype)
+			return returnResponse, nil
+		}
+		// we need to read all the body now, close it, and replace it with another reader
+		// as there is currently no way of "resetting" a Body
+		body, err := ioutil.ReadAll(returnResponse.Body)
+		if err != nil {
+			fmt.Printf("error reading cached response body: %s\n", err.Error())
+			return returnResponse, nil
+		}
+		returnResponse.Body.Close()
+		var rangedRequestStart, rangedRequestEnd int64
+		//TODO(uovobw): handle corrupted/nonstandard request header
+		rangeList := strings.Split(rangevalue, rangeSeparator)
```

Comment:
It's not necessary (for our needs) and probably not as useful to support multiple byte range requests (e.g., specifying multiple ranges with comma separators; see http://greenbytes.de/tech/webdav/draft-ietf-httpbis-p5-range-latest.html#rule.ranges-specifier), but it should not produce incorrect behavior in that case. (Currently it would fail to parse the ints because they would include commas, and then it would return a 0-range response, I believe.)